# HG changeset patch
# User Torsten <ttl@justmail.de>
# Date 1392326264 -3600
#      Thu Feb 13 22:17:44 2014 +0100
# Branch gui-release
# Node ID f959c63934e6c9485f002117c35db3c7f7c3760c
# Parent  f01ac1bb8a5d97f0899ce7250564d43db64d3f77
Backed out changeset f01ac1bb8a5d which broke some editor short cuts

diff --git a/libgui/src/m-editor/file-editor-tab.cc b/libgui/src/m-editor/file-editor-tab.cc
--- a/libgui/src/m-editor/file-editor-tab.cc
+++ b/libgui/src/m-editor/file-editor-tab.cc
@@ -762,47 +762,16 @@ file_editor_tab::unindent_selected_text 
   if (ID != this)
     return;
 
   do_indent_selected_text (false);
 }
 
 
 void
-file_editor_tab::zoom_in (const QWidget *ID)
-{
-  if (ID != this)
-    return;
-
-  _edit_area->zoomIn (1);
-  auto_margin_width ();
-}
-
-void
-file_editor_tab::zoom_out (const QWidget *ID)
-{
-  if (ID != this)
-    return;
-
-  _edit_area->zoomOut (1);
-  auto_margin_width ();
-}
-
-void
-file_editor_tab::zoom_normal (const QWidget *ID)
-{
-  if (ID != this)
-    return;
-
-  _edit_area->zoomTo (0);
-  auto_margin_width ();
-}
-
-
-void
 file_editor_tab::handle_find_dialog_finished (int)
 {
   // Find dialog is going to hide.  Save location of window for
   // when it is reshown.
   _find_dialog_geometry = _find_dialog->geometry ();
   _find_dialog_is_visible = false;
 }
 
diff --git a/libgui/src/m-editor/file-editor-tab.h b/libgui/src/m-editor/file-editor-tab.h
--- a/libgui/src/m-editor/file-editor-tab.h
+++ b/libgui/src/m-editor/file-editor-tab.h
@@ -92,20 +92,16 @@ public slots:
   void remove_all_breakpoints (const QWidget *ID);
 
   void comment_selected_text (const QWidget *ID);
   void uncomment_selected_text (const QWidget *ID);
 
   void indent_selected_text (const QWidget *ID);
   void unindent_selected_text (const QWidget *ID);
 
-  void zoom_in (const QWidget *ID);
-  void zoom_out (const QWidget *ID);
-  void zoom_normal (const QWidget *ID);
-
   void find (const QWidget *ID);
   void goto_line (const QWidget *ID, int line = -1);
 
   void insert_debugger_pointer (const QWidget *ID, int line = -1);
   void delete_debugger_pointer (const QWidget *ID, int line = -1);
 
   void do_breakpoint_marker (bool insert, const QWidget *ID, int line = -1);
 
diff --git a/libgui/src/m-editor/file-editor.cc b/libgui/src/m-editor/file-editor.cc
--- a/libgui/src/m-editor/file-editor.cc
+++ b/libgui/src/m-editor/file-editor.cc
@@ -820,34 +820,16 @@ file_editor::handle_add_filename_to_list
 
 void
 file_editor::active_tab_changed (int index)
 {
   emit fetab_change_request (_tab_widget->widget (index));
 }
 
 void
-file_editor::zoom_in (bool)
-{
-  emit fetab_zoom_in (_tab_widget->currentWidget ());
-}
-
-void
-file_editor::zoom_out (bool)
-{
-  emit fetab_zoom_out (_tab_widget->currentWidget ());
-}
-
-void
-file_editor::zoom_normal (bool)
-{
-  emit fetab_zoom_normal (_tab_widget->currentWidget ());
-}
-
-void
 file_editor::handle_editor_state_changed (bool copy_available,
                                           const QString& file_name)
 {
   // In case there is some scenario where traffic could be coming from
   // all the file editor tabs, just process info from the current active tab.
   if (sender () == _tab_widget->currentWidget ())
     {
       _copy_action->setEnabled (copy_available);
@@ -1126,25 +1108,16 @@ file_editor::construct (void)
                          tr ("&Preferences..."),
                          this, SLOT (request_preferences (bool)));
   _styles_preferences_action =
     editMenu->addAction (QIcon (":/actions/icons/configure.png"),
                          tr ("&Styles Preferences..."),
                          this, SLOT (request_styles_preferences (bool)));
   _menu_bar->addMenu (editMenu);
 
-  QMenu *view_menu = new QMenu (tr ("&View"), _menu_bar);
-  _zoom_in_action = view_menu->addAction (QIcon (), tr ("Zoom &In"),
-                                this, SLOT (zoom_in (bool)));
-  _zoom_out_action = view_menu->addAction (QIcon (), tr ("Zoom &Out"),
-                                this, SLOT (zoom_out (bool)));
-  _zoom_normal_action = view_menu->addAction (QIcon (), tr ("&Normal Size"),
-                                this, SLOT (zoom_normal (bool)));
-  _menu_bar->addMenu (view_menu);
-
   _debug_menu = new QMenu (tr ("&Debug"), _menu_bar);
   _debug_menu->addAction (toggle_breakpoint_action);
   _debug_menu->addAction (next_breakpoint_action);
   _debug_menu->addAction (previous_breakpoint_action);
   _debug_menu->addAction (remove_all_breakpoints_action);
   _debug_menu->addSeparator ();
   // The other debug actions will be added by the main window.
   _menu_bar->addMenu (_debug_menu);
@@ -1355,23 +1328,16 @@ file_editor::add_file_editor_tab (file_e
            f, SLOT (copy (const QWidget*)));
 
   connect (this, SIGNAL (fetab_cut (const QWidget*)),
            f, SLOT (cut (const QWidget*)));
 
   connect (this, SIGNAL (fetab_paste (const QWidget*)),
            f, SLOT (paste (const QWidget*)));
 
-  connect (this, SIGNAL (fetab_zoom_in (const QWidget*)),
-           f, SLOT (zoom_in (const QWidget*)));
-  connect (this, SIGNAL (fetab_zoom_out (const QWidget*)),
-           f, SLOT (zoom_out (const QWidget*)));
-  connect (this, SIGNAL (fetab_zoom_normal (const QWidget*)),
-           f, SLOT (zoom_normal (const QWidget*)));
-
   connect (this, SIGNAL (fetab_context_help (const QWidget*, bool)),
            f, SLOT (context_help (const QWidget*, bool)));
 
   connect (this, SIGNAL (fetab_context_edit (const QWidget*)),
            f, SLOT (context_edit (const QWidget*)));
 
   connect (this, SIGNAL (fetab_save_file (const QWidget*)),
            f, SLOT (save_file (const QWidget*)));
@@ -1485,21 +1451,16 @@ file_editor::set_shortcuts (bool set)
                                                 + Qt::Key_Tab);
 
       _copy_action->setShortcut (QKeySequence::Copy);
       _cut_action->setShortcut (QKeySequence::Cut);
       _paste_action->setShortcut (QKeySequence::Paste);
       _context_help_action->setShortcut (QKeySequence::HelpContents);
       _context_doc_action->setShortcut (Qt::SHIFT + Qt::Key_F1);
 
-      _zoom_in_action->setShortcut (QKeySequence::ZoomIn);
-      _zoom_out_action->setShortcut (QKeySequence::ZoomOut);
-      _zoom_normal_action->setShortcut (Qt::ControlModifier
-                                         + Qt::AltModifier + Qt::Key_0);
-
       _find_action->setShortcut (QKeySequence::Find);
       _goto_line_action->setShortcut (Qt::ControlModifier+ Qt::Key_G);
 
       _next_bookmark_action->setShortcut (Qt::Key_F2);
       _previous_bookmark_action->setShortcut (Qt::SHIFT + Qt::Key_F2);
       _toggle_bookmark_action->setShortcut (Qt::Key_F7);
 
       _print_action->setShortcut (QKeySequence::Print);
@@ -1524,20 +1485,16 @@ file_editor::set_shortcuts (bool set)
       _indent_selection_action->setShortcut (no_key);
       _unindent_selection_action->setShortcut (no_key);
 
       _copy_action->setShortcut (no_key);
       _cut_action->setShortcut (no_key);
       _paste_action->setShortcut (no_key);
       _context_help_action->setShortcut (no_key);
 
-      _zoom_in_action->setShortcut (no_key);
-      _zoom_out_action->setShortcut (no_key);
-      _zoom_normal_action->setShortcut (no_key);
-
       _find_action->setShortcut (no_key);
       _goto_line_action->setShortcut (no_key);
 
       _next_bookmark_action->setShortcut (no_key);
       _previous_bookmark_action->setShortcut (no_key);
       _toggle_bookmark_action->setShortcut (no_key);
 
       _print_action->setShortcut (no_key);
@@ -1564,20 +1521,16 @@ file_editor::check_actions ()
 
   _indent_selection_action->setEnabled (have_tabs);
   _unindent_selection_action->setEnabled (have_tabs);
 
   _paste_action->setEnabled (have_tabs);
   _context_help_action->setEnabled (have_tabs);
   _context_doc_action->setEnabled (have_tabs);
 
-  _zoom_in_action->setEnabled (have_tabs);
-  _zoom_out_action->setEnabled (have_tabs);
-  _zoom_normal_action->setEnabled (have_tabs);
-
   _find_action->setEnabled (have_tabs);
   _goto_line_action->setEnabled (have_tabs);
 
   _next_bookmark_action->setEnabled (have_tabs);
   _previous_bookmark_action->setEnabled (have_tabs);
   _toggle_bookmark_action->setEnabled (have_tabs);
   _remove_bookmark_action->setEnabled (have_tabs);
 
diff --git a/libgui/src/m-editor/file-editor.h b/libgui/src/m-editor/file-editor.h
--- a/libgui/src/m-editor/file-editor.h
+++ b/libgui/src/m-editor/file-editor.h
@@ -97,21 +97,16 @@ signals:
   void fetab_unindent_selected_text (const QWidget* ID);
   void fetab_find (const QWidget* ID);
   void fetab_goto_line (const QWidget* ID, int line = -1);
   void fetab_insert_debugger_pointer (const QWidget* ID, int line = -1);
   void fetab_delete_debugger_pointer (const QWidget* ID, int line = -1);
   void fetab_do_breakpoint_marker (bool insert, const QWidget* ID,
                                    int line = -1);
   void fetab_set_focus (const QWidget* ID);
-
-  void fetab_zoom_in (const QWidget* ID);
-  void fetab_zoom_out (const QWidget* ID);
-  void fetab_zoom_normal (const QWidget* ID);
-
   void request_settings_dialog (const QString&);
   void execute_command_in_terminal_signal (const QString&);
   void file_loaded_signal ();
 
 public slots:
   void focus (void);
 
   void request_new_file (const QString& commands);
@@ -190,20 +185,16 @@ private slots:
 
   void request_open_file (const QString& fileName, int line = -1,
                           bool debug_pointer = false,
                           bool breakpoint_marker = false, bool insert = true);
   void request_preferences (bool);
   void request_styles_preferences (bool);
   void restore_create_file_setting ();
 
-  void zoom_in (bool);
-  void zoom_out (bool);
-  void zoom_normal (bool);
-
 private:
 
   void construct (void);
   void add_file_editor_tab (file_editor_tab *f, const QString& fn);
   void save_file_as (QWidget *fetabID = 0);
   void mru_menu_update (void);
   bool call_custom_editor (const QString& file_name = QString (), int line = -1);
 
@@ -224,20 +215,16 @@ private:
   QAction *_unindent_selection_action;
 
   QAction *_copy_action;
   QAction *_cut_action;
   QAction *_paste_action;
   QAction *_context_help_action;
   QAction *_context_doc_action;
 
-  QAction *_zoom_in_action;
-  QAction *_zoom_out_action;
-  QAction *_zoom_normal_action;
-
   QAction *_find_action;
   QAction *_goto_line_action;
 
   QAction *_next_bookmark_action;
   QAction *_previous_bookmark_action;
   QAction *_toggle_bookmark_action;
   QAction * _remove_bookmark_action;
 
diff --git a/libgui/src/m-editor/octave-qscintilla.cc b/libgui/src/m-editor/octave-qscintilla.cc
--- a/libgui/src/m-editor/octave-qscintilla.cc
+++ b/libgui/src/m-editor/octave-qscintilla.cc
@@ -24,26 +24,23 @@ along with Octave; see the file COPYING.
 
 #ifdef HAVE_CONFIG_H
 #include <config.h>
 #endif
 
 #ifdef HAVE_QSCINTILLA
 
 #include <Qsci/qscilexer.h>
-#include <QShortcut>
 
 #include "octave-qscintilla.h"
 #include "file-editor-tab.h"
 
 octave_qscintilla::octave_qscintilla (QWidget *p)
   : QsciScintilla (p)
-{
-  installEventFilter (this);
-}
+{ }
 
 octave_qscintilla::~octave_qscintilla ()
 { }
 
 void
 octave_qscintilla::get_global_textcursor_pos (QPoint *global_pos,
                                               QPoint *local_pos)
 {
@@ -184,18 +181,9 @@ void
 octave_qscintilla::contextmenu_run (bool)
 {
   QStringList commands = selectedText ().split (QRegExp("[\r\n]"),
                                                 QString::SkipEmptyParts);
   for (int i = 0; i < commands.size (); i++ )
     emit execute_command_in_terminal_signal (commands.at (i));
 }
 
-bool
-octave_qscintilla::eventFilter(QObject *obj, QEvent *e)
- {
-  if (e->type() == QEvent::ShortcutOverride)
-    return true;
-  else
-    return QObject::eventFilter(obj, e);
- }
-
 #endif
diff --git a/libgui/src/m-editor/octave-qscintilla.h b/libgui/src/m-editor/octave-qscintilla.h
--- a/libgui/src/m-editor/octave-qscintilla.h
+++ b/libgui/src/m-editor/octave-qscintilla.h
@@ -57,13 +57,12 @@ private slots:
   void contextmenu_doc (bool);
   void contextmenu_help_doc (bool);
   void contextmenu_edit (bool);
   void contextmenu_run (bool);
 
 private:
 
   QString _word_at_cursor;
-  bool eventFilter(QObject *obj, QEvent *e);
 
 };
 
 #endif
